plugin/src/test/groovy/org/gradle/testretry/AbstractConfigCacheFuncTest.groovy

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.testretry
+
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.util.GradleVersion
+import org.junit.Assume
+import org.spockframework.util.VersionNumber
+import spock.lang.Unroll
+
+abstract class AbstractConfigCacheFuncTest extends AbstractGeneralPluginFuncTest {
```

Comment:
We don't need a separate hierarchy for this. 

Please find a way to incorporate this into the existing hierarchy. You might need to add a new base class for the framework specific tests.

Reply:
Good point.  I think I expected it to diverge more when I started this.  I've merged these classes with the existing functional test classes and created a base framework class.  Let me know if you think we need to do more here.